packages/components/file-card/src/file-card.js

```diff
@@ -0,0 +1,357 @@
+				uploading: 'focus',
+				download: 'default',
+				failed: 'failed'
+			};
+			return cardTypes[this.state];
+		}
+
+		get errorMessages() {
+			if (this.state === 'failed') {
+				const message = this.errorMessage;
+				return html`
+					<ts-typography
+						class="error-message"
+						text="${message}"
+						color="error"
+						variant="subtitle"
+						no-wrap
+					></ts-typography>
+				`;
+			}
+		}
+
+		get fileIcon() {
+			return html`
+				<span class="file-icon-wrapper icon-wrapper">
+					<!--TODO: use icon components-->
+					<svg
+						class="file-icon"
+						width="32px"
+						height="42px"
+						viewBox="0 0 32 42"
+						version="1.1"
+						xmlns="http://www.w3.org/2000/svg"
+						xmlns:xlink="http://www.w3.org/1999/xlink"
+					>
+						<g id="Symbols" stroke-width="1">
+							<g
+								id="card/5.-File-card/mouseover"
+								transform="translate(-9.000000, -9.000000)"
+							>
+								<g id="5.-File-card">
+									<g id="File-icon" transform="translate(10.000000, 10.000000)">
+										<path
+											d="M19.5667631,-0.5 L29.5322581,9.79767818 L29.5322581,38.0952381 C29.5322581,39.3654698 28.5111819,40.5 27.3244782,40.5 L1.70777989,40.5 C0.521076133,40.5 -0.5,39.3654698 -0.5,38.0952381 L-0.5,1.9047619 C-0.5,0.634530207 0.521076133,-0.5 1.70777989,-0.5 L19.5667631,-0.5 Z"
+											id="outline"
+										></path>
+										<path
+											d="M19.3548387,0 L29.9943074,11 L21.3186802,11 C20.2340805,11 19.3548387,10.0430649 19.3548387,8.8660465 L19.3548387,0 L19.3548387,0 Z"
+											id="fold"
+										></path>
+									</g>
+								</g>
+							</g>
+						</g>
+					</svg>
+				</span>
+			`;
+		}
+
+		get progressBar() {
+			if (this.state === 'uploading') {
+				return html`
+					<ts-progress-bar class="progress-bar" indeterminate></ts-progress-bar>
+				`;
+			}
+		}
+
+		get actionIcons() {
+			return html`
+				<div>
+					<span class="remove-action action-icon-wrapper icon-wrapper">
+						<svg
+							class="action-icon"
+							width="16px"
+							height="16px"
+							viewBox="0 0 16 16"
+							version="1.1"
+							xmlns="http://www.w3.org/2000/svg"
+							xmlns:xlink="http://www.w3.org/1999/xlink"
+						>
+							<g id="Symbols" stroke="none" stroke-width="1">
+								<g
+									id="card/5.-File-card/standard"
+									transform="translate(-262.000000, -22.000000)"
+								>
+									<g id="5.-File-card">
+										<g
+											id="icon/UI-library/clear"
+											transform="translate(258.000000, 18.000000)"
+										>
+											<path
+												d="M4,12 C4,7.581722 7.59071231,4 12,4 C16.418278,4 20,7.59071231 20,12 C20,16.418278 16.4092877,20 12,20 C7.581722,20 4,16.4092877 4,12 Z M12,13.1313708 L15.0343146,16.1656854 L16.1656854,15.0343146 L13.1313708,12 L16.1656854,8.96568542 L15.0343146,7.83431458 L12,10.8686292 L8.96568542,7.83431458 L7.83431458,8.96568542 L10.8686292,12 L7.83431458,15.0343146 L8.96568542,16.1656854 L12,13.1313708 Z"
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 412 characters long line (which is greater than 120 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3ALineLength)